scripts/asm-installer/install_asm

```diff
@@ -1977,6 +1982,58 @@ EOF
   outro
 }
 
+pref_len() {
+  local LEN; LEN="${#1}";
+  if ((${#2} > $LEN)); then
+    LEN="${#2}"
+  fi
+
+  for ((i=0; i<$LEN; ++i)); do
+    if [[ "${1:i:1}" != "${2:i:1}" ]]; then
+      echo "${i}"
+      return
+    fi
+  done
+
+  echo "${LEN}"
+}
+
+find_best_region() {
+  REGION="global"
+  BEST=0
+  for location in $(gcloud beta logging locations list --format='value(location_id)'); do
+    SCORE="$(pref_len "${CLUSTER_LOCATION}" "${location}")"
+    if (( $SCORE > $BEST )); then
+      BEST="${SCORE}"
+      REGION="${location}"
+    fi
+  done
+
+  echo ${REGION}
+}
```

Comment:
Well now I know I can't write pretty bash:

```
In install_asm line 1987:
  if ((${#2} > $LEN)); then
               ^-- SC2004: $/${} is unnecessary on arithmetic variables.


In install_asm line 1991:
  for ((i=0; i<$LEN; ++i)); do
               ^-- SC2004: $/${} is unnecessary on arithmetic variables.


In install_asm line 2006:
    if (( $SCORE > $BEST )); then
          ^-- SC2004: $/${} is unnecessary on arithmetic variables.
                   ^-- SC2004: $/${} is unnecessary on arithmetic variables.


In install_asm line 2012:
  echo ${REGION}
       ^-- SC2086: Double quote to prevent globbing and word splitting.

```

Reply:
Updated per these recommendations.